%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[letterpaper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{2}



\title{CopyNumberVariantsSequenceAnalysis}
\date{Jul 14, 2022}
\release{1.0}
\author{Michael Hall}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Version 1.0}
\makeindex
\begin{document}

\pagestyle{empty}

        \pagenumbering{Roman} %%% to avoid page 1 conflict with actual page
        \begin{titlepage}
            \vspace*{10mm} %%% * is used to give space from top
            \flushright\textbf{\Huge {CopyNumberVariantsSequenceAnalysis\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {A Step-by-Step Guide\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {[DRAFT]\\}}
            \vspace{50mm}
            \textbf{\Large {Michael Hall\\}}
            \vspace{10mm}
            \textbf{\Large {Plant Breeding and Genetics Laboratory\\}}
            \vspace{0mm}
            \textbf{\Large {FAO/IAEA Joint Division\\}}
            \vspace{0mm}
            \textbf{\Large {Seibersdorf, Austria\\}}
	    \vspace{10mm}
            \normalsize Created: July, 2022\\
            \vspace*{0mm}
            \normalsize  Last updated: 14 July 2022
            %% \vfill adds at the bottom
            \vfill
            \small\flushleft {{\textbf {Please note:}} \textit {This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views
expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.}}
        \end{titlepage}
        \pagenumbering{arabic}
        \newcommand{\sectionbreak}{\clearpage}

\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\section{CNVseq Analysis Banana and Sorghum}
\label{\detokenize{CNV:cnvseq-analysis-banana-and-sorghum}}\label{\detokenize{CNV::doc}}

\bigskip\hrule\bigskip



\subsection{Software Prerequisites}
\label{\detokenize{CNV:software-prerequisites}}
\sphinxAtStartPar
\#Burrows\sphinxhyphen{}Wheeler\sphinxhyphen{}Aligner
(\sphinxurl{http://bio-bwa.sourceforge.net/)(see} line 126).
\#Download and Install BBmap
\sphinxurl{https://sourceforge.net/projects/bbmap/}
Bin\sphinxhyphen{}by\sphinxhyphen{}Sam\sphinxhyphen{}tool (see github repository)
Python version 2.7(See enivornment .yaml)

\sphinxAtStartPar
\sphinxstylestrong{Banana}

\sphinxAtStartPar
\sphinxstylestrong{Procure your raw FASTQ reads from NCBI of two Banana samples, one is a known mutant Novaria and the other is a wildtype Naine and follow the protocol.}
\sphinxstylestrong{Efficient Screening Techniques to Identify Mutants with TR4 Resistance in Banana}
p.117 \sphinxhyphen{} 127
\sphinxstylestrong{Use clumpify script to remove duplicates}

\sphinxAtStartPar
(\sphinxurl{https://www.ncbi.nlm.nih.gov/bioproject/PRJNA627139})

\sphinxAtStartPar
\#Download sratools

\begin{sphinxVerbatim}[commandchars=\\\{\}]
srapath SRR11579627

prefetch SRR11579627

wget https://sra\PYGZhy{}downloadb.be\PYGZhy{}md.ncbi.nlm.nih.gov/sos3/sra\PYGZhy{}pub\PYGZhy{}run\PYGZhy{}21/SRR11579627/SRR11579627.1

\PYG{c+c1}{\PYGZsh{}Convert SRA into fastq}

fastq\PYGZhy{}dump \PYGZhy{}\PYGZhy{}split\PYGZhy{}3 SRR11579627


srapath SRR11579628

prefetch SRR11579628

wget https://sra\PYGZhy{}downloadb.be\PYGZhy{}md.ncbi.nlm.nih.gov/sos3/sra\PYGZhy{}pub\PYGZhy{}run\PYGZhy{}21/SRR11579628/SRR11579628.1

\PYG{c+c1}{\PYGZsh{}Convert SRA into fastq}

fastq\PYGZhy{}dump \PYGZhy{}\PYGZhy{}split\PYGZhy{}3 SRR11579628
\end{sphinxVerbatim}


\subsection{Rename FASTQ}
\label{\detokenize{CNV:rename-fastq}}\begin{quote}

\sphinxAtStartPar
\sphinxstylestrong{Naine.R1.fq.gz}
\sphinxstylestrong{Naine.R2.fq.gz}
\sphinxstylestrong{Novaria.R1.fq.gz}
\sphinxstylestrong{Novaria.R2.fq.gz}

\sphinxAtStartPar
\sphinxstylestrong{Run the clumpify python script to remove duplicates per sample.}
\end{quote}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
./clumpify.sh \PYG{k}{in}\PYG{o}{=}Naine.R1.fq.gz \PYG{n+nv}{in2}\PYG{o}{=}Naine.R2.fq.gz \PYG{n+nv}{out}\PYG{o}{=}Naine.R1.
dedup.fastq.gz \PYG{n+nv}{out2}\PYG{o}{=}Naine.R2.dedup.fastq.gz \PYG{n+nv}{dedupe}\PYG{o}{=}t

./clumpify.sh \PYG{k}{in}\PYG{o}{=}Novaria.R1.fq.gz \PYG{n+nv}{in2}\PYG{o}{=}Novaria.R2.fq.gz \PYG{n+nv}{out}\PYG{o}{=}Novaria.R1.
dedup.fastq.gz \PYG{n+nv}{out2}\PYG{o}{=}Novaria.R2.dedup.fastq.gz \PYG{n+nv}{dedupe}\PYG{o}{=}t
\end{sphinxVerbatim}


\subsection{Standard Output Clumpify python}
\label{\detokenize{CNV:standard-output-clumpify-python}}\begin{quote}

\sphinxAtStartPar
Done!
Time:                           31.447 seconds.
Reads Processed:         6262k  199.16k reads/sec
Bases Processed:         1885m  59.94m bases/sec

\sphinxAtStartPar
Reads In:              6262958
Clumps Formed:         1730359
Duplicates Found:         3782
Reads Out:             6259176
Bases Out:          1884185686
Total time:     51.345 seconds.

\sphinxAtStartPar
\sphinxstylestrong{NOVARIA}

\sphinxAtStartPar
Done!
Time:                           29.438 seconds.
Reads Processed:         6000k  203.82k reads/sec
Bases Processed:         1837m  62.43m bases/sec

\sphinxAtStartPar
Reads In:              6000036
Clumps Formed:         1648176
Duplicates Found:         2026
Reads Out:             5998010
Bases Out:          1837286910
Total time:     50.222 seconds.
\end{quote}


\bigskip\hrule\bigskip



\subsection{Download Reference Genome NCBI}
\label{\detokenize{CNV:download-reference-genome-ncbi}}
\sphinxAtStartPar
\sphinxurl{https://www.ncbi.nlm.nih.gov/assembly/GCF\_000313855.2}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
mkdir BananaGamma
mv Novaria.R1.dedup.fastq.gz Novaria.R2.dedup.fastq.gz BananaGamma/
mv Naine.R1.dedup.fastq.gz Naine.R2.dedup.fastq.gz BananaGamma/
\PYG{n+nb}{cd} BananaGamma

mkdir Genome
mv *.fna Genome/
\PYG{n+nb}{cd} Genome bwa index *.fna

\PYG{n+nb}{cd} ../

https://github.com/lh3/bwa

git clone https://github.com/lh3/bwa.git
\PYG{n+nb}{cd} bwa\PYG{p}{;} make
./bwa
\PYG{c+c1}{\PYGZsh{}Needs to be Harvard Version}

./bwa mem \PYGZhy{}M \PYGZhy{}t \PYG{l+m}{4} ../Genome/*.fna Novaria.R2.dedup.fq Novaria.R2.dedup.fq \PYGZgt{} Novaria.dedup.sam

./bwa mem \PYGZhy{}M \PYGZhy{}t \PYG{l+m}{4} Genome/*.fna Naine.R1.dedup.fastq.gz Naine.R2.dedup.fastq.gz \PYGZgt{} Naine.dedup.sam

samtools sort \PYGZhy{}O sam \PYGZhy{}T sam \PYGZhy{}T Novaria.sort \PYGZhy{}o Novaria\PYGZus{}aln.sam Novaria.dedup.sam
samtools sort \PYGZhy{}O sam \PYGZhy{}T sam \PYGZhy{}T Naine.sort \PYGZhy{}o Naine\PYGZus{}aln.sam Naine.dedup.sam

samtools view \PYGZhy{}b Novaria.dedup.sam \PYGZgt{} Novaria.bam
samtools view \PYGZhy{}b Naine.dedup.sam \PYGZgt{} Naine.bam

samtools index Novaria.bam
samtools index Naine.bam

mv Novaria\PYGZus{}aln.sam Naine\PYGZus{}aln.sam Bin\PYGZhy{}by\PYGZhy{}Sam\PYGZhy{}tool/
\PYG{n+nb}{cd} Bin\PYGZhy{}by\PYGZhy{}Sam\PYGZhy{}tool
python bin\PYGZhy{}by\PYGZhy{}sam\PYGZus{}2.0.py \PYGZhy{}o N3\PYGZus{}100kbin.txt \PYGZhy{}s \PYG{l+m}{100000} \PYGZhy{}b \PYGZhy{}p \PYG{l+m}{3} \PYGZhy{}c Naine\PYGZus{}.aln.sam
\end{sphinxVerbatim}


\subsection{Download the r package rom PBGLMichael/CNVseq repository}
\label{\detokenize{CNV:download-the-r-package-rom-pbglmichael-cnvseq-repository}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Banana CNV}

setwd\PYG{o}{(}\PYG{l+s+s2}{\PYGZdq{}/home/michael/Desktop/Banana/Banana\PYGZus{}LC\PYGZus{}WGS\PYGZdq{}}\PYG{o}{)}
devtools::install\PYGZus{}github\PYG{o}{(}\PYG{n+nv}{repo} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}PBGLMichaelHall/CNVseq\PYGZdq{}},force \PYG{o}{=} TRUE\PYG{o}{)}
library\PYG{o}{(}CNV\PYG{o}{)}

CNV::CNV\PYG{o}{(}\PYG{n+nv}{file} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}N3\PYGZus{}100kbin.txt\PYGZdq{}},Chromosome \PYG{o}{=}  c\PYG{o}{(}\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025202.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025203.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025203.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025204.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025205.1\PYGZdq{}},
\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025206.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025207.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025208.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025209.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025210.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025211.1\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025212.1\PYGZdq{}}\PYG{o}{)},
\PYG{n+nv}{mutantname} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}Novaria.Naine\PYGZdq{}},controlname \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}Naine.Naine\PYGZdq{}},size \PYG{o}{=} .75,alpha \PYG{o}{=} .25,color\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}green\PYGZdq{}}\PYG{o}{)}
\end{sphinxVerbatim}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{777}.png}
\end{figure}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{header}.png}
\end{figure}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{ge}.png}
\end{figure}


\subsection{Chromosome 5}
\label{\detokenize{CNV:chromosome-5}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
CNV::CNV\PYG{o}{(}\PYG{n+nv}{file} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}N3\PYGZus{}100kbin.txt\PYGZdq{}},Chromosome \PYG{o}{=}  c\PYG{o}{(}\PYG{l+s+s2}{\PYGZdq{}NC\PYGZus{}025206.1\PYGZdq{}}\PYG{o}{)},mutantname \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}Novaria.Naine\PYGZdq{}},
\PYG{n+nv}{controlname} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}Naine.Naine\PYGZdq{}},size \PYG{o}{=} .75,alpha \PYG{o}{=} .25,color\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}green\PYGZdq{}}\PYG{o}{)}
\end{sphinxVerbatim}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{Rplot777}.png}
\end{figure}



\begin{sphinxVerbatim}[commandchars=\\\{\}]
samtools view \PYGZhy{}h con\PYGZhy{}2\PYGZus{}S1\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09.bam \PYGZgt{} con\PYGZhy{}2\PYGZus{}S1\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09\PYGZus{}aln.sam

**MUTANT**
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
samtools view \PYGZhy{}h D2\PYGZhy{}1\PYGZus{}S7\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09.bam \PYGZgt{} D2\PYGZhy{}1\PYGZus{}S7\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09\PYGZus{}aln.sam
\end{sphinxVerbatim}


\subsection{bin\sphinxhyphen{}by\sphinxhyphen{}sam\_2.0.py python script}
\label{\detokenize{CNV:bin-by-sam-2-0-py-python-script}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{\PYGZdl{}python} bin\PYGZhy{}by\PYGZhy{}sam\PYGZus{}2.0.py \PYGZhy{}o N3\PYGZus{}100kbin.txt \PYGZhy{}s \PYG{l+m}{100000} \PYGZhy{}b \PYGZhy{}p \PYG{l+m}{3} \PYGZhy{}c con\PYGZhy{}2\PYGZus{}S1\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09\PYGZus{}aln.sam
\end{sphinxVerbatim}


\subsection{\sphinxstylestrong{Sorghum CNV}}
\label{\detokenize{CNV:sorghum-cnv}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
CNV::CNV\PYG{o}{(}\PYG{n+nv}{file} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}N3\PYGZus{}100kbin.txt\PYGZdq{}},Chromosome \PYG{o}{=} c\PYG{o}{(}\PYG{l+s+s2}{\PYGZdq{}Chr04\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}Chr05\PYGZdq{}},\PYG{l+s+s2}{\PYGZdq{}Chr09\PYGZdq{}}\PYG{o}{)},
\PYG{n+nv}{mutantname} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}con.2.NA\PYGZdq{}},controlname \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}D2.2.NA\PYGZdq{}},size \PYG{o}{=} .75,alpha \PYG{o}{=} \PYG{l+m}{5}.0,color\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}green\PYGZdq{}}\PYG{o}{)}
\end{sphinxVerbatim}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{SorghumCNV7}.png}
\end{figure}


\subsection{Chromosome 9}
\label{\detokenize{CNV:chromosome-9}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
CNV::CNV\PYG{o}{(}\PYG{n+nv}{file}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}N3\PYGZus{}100kbin.txt\PYGZdq{}},Chromosome\PYG{o}{=}c\PYG{o}{(}\PYG{l+s+s2}{\PYGZdq{}Chr09\PYGZdq{}}\PYG{o}{)},mutantname\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}con.2.NA\PYGZdq{}},controlname\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}D2.2.NA\PYGZdq{}},size\PYG{o}{=}.75,alpha\PYG{o}{=}\PYG{l+m}{5}.0\PYG{o}{)}
\end{sphinxVerbatim}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{Rplot01}.png}
\end{figure}



\renewcommand{\indexname}{Index}
\printindex
\end{document}